src/packages/core/src/resolvers.ts

```diff
@@ -0,0 +1,475 @@
+				...hookParams,
+				deleted: success,
+			}));
+
+		return success;
+	};
+
+export const deleteMany =
+	(entity: EntityMetadata<any, any>) =>
+	async <G extends WithId & { name: string }, C extends BaseContext>(
+		_source: unknown,
+		{ filter }: { filter: Filter<G> },
+		context: C,
+		info: GraphQLResolveInfo
+	) => {
+		if (!entity.provider) {
+			throw new Error(
+				`Entity ${entity.name} does not have a provider, cannot resolve delete operation.`
+			);
+		}
+		return withTransaction(entity.provider, async () => {
+			if (!entity.provider?.deleteMany) throw new Error('Provider has not implemented DeleteMany.');
+
+			const hookManager = hookManagerMap.get(entity.name);
+			const params: DeleteManyHookParams<G> = {
+				args: { filter },
+				info,
+				context,
+				transactional: !!entity.provider.withTransaction,
+			};
+
+			const hookParams = hookManager
+				? await hookManager.runHooks(HookRegister.BEFORE_DELETE, params)
+				: params;
+
+			if (!hookParams.args?.filter) throw new Error('No delete ids specified cannot continue.');
+
+			const success = await entity.provider.deleteMany(hookParams.args?.filter);
+
+			hookManager &&
+				(await hookManager.runHooks(HookRegister.AFTER_DELETE, {
+					...hookParams,
+					deleted: success,
+				}));
+
+			return success;
+		});
+	};
+
+export const listRelationshipField = async <
+	G extends WithId & { name: string } & { dataEntity: D },
+	D extends BaseDataEntity,
+	C extends BaseContext,
+>(
+	source: G,
+	input: { filter: Filter<G> },
+	context: C,
+	info: GraphQLResolveInfo
+) => {
+	logger.trace(`Resolving ${info.parentType.name}.${info.fieldName}`);
+
+	if (!info.path.typename)
+		throw new Error(`No typename found in path for ${info.path}, this should not happen.`);
+
+	const entity = graphweaverMetadata.getEntityByName(info.path.typename);
+	if (!entity) {
+		throw new Error(`Entity ${source.name} not found in metadata. This should not happen.`);
+	}
+
+	// If we've already resolved the data, we may want to return it, but we want the hooks to run first.
+	const existingData = source[info.fieldName as keyof typeof source];
+
+	const field = entity.fields[info.fieldName];
+	const { id, relatedField } = field.relationshipInfo ?? {};
+	const idValue = !id
+		? undefined
+		: typeof id === 'function'
+			? id(source.dataEntity)
+			: (source.dataEntity as any)[id];
+
+	if (typeof existingData === 'undefined' && !idValue && !field.relationshipInfo?.relatedField) {
+		// id is null and we are loading a single instance so let's return null
+		return null;
+	}
+
+	let gqlEntityType = field.getType() as
+		| GraphQLEntityConstructor<G, D>
+		| GraphQLEntityConstructor<G, D>[];
+	let isList = false;
+
+	if (Array.isArray(gqlEntityType)) {
+		isList = true;
+		gqlEntityType = gqlEntityType[0];
+	}
+
+	// @todo: Should the user specified filter be and-ed here?
+	//        My worry is if we just pass the filter through, it could be used to circumvent the relationship join.
```

Comment:
Should we start by anding? That way we know our filter is always applied, I think that is how I assumed it would work when using it as you can potentially get data that is not related to the parent.

Reply:
Yeah I'm down with that.

Reply:
done 🚀